src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package com.synacy.lesson02.exercises;
+
+import com.synacy.lesson02.exercises.domain.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.xml.ws.soap.MTOM;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+
+    private EnrollmentService enrollmentService;
+    private EnrollmentDto enrollmentDetails;
+
+    @Mock StudentProfileService studentProfileService;
+    @Mock CourseClassService courseClassService;
+    @Mock EnrollmentNotificationService enrollmentNotificationService;
+    @Mock StudyLoadFormatter studyLoadFormatter;
+    @Mock SystemService systemService;
+
+    @Before
+    public void setup(){
+        enrollmentService = new EnrollmentService();
+
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+        enrollmentService.setSystemService(systemService);
+        enrollmentDetails = mock(EnrollmentDto.class);
+
+    }
+
+    @Test
+    public void processEnrollment_shouldUpdateStudentProfileWithEnrollmentDetails() throws Exception {
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+        verify(studentProfileService).updateStudentProfile(enrollmentDetails.getStudentProfile());
```

Comment:
Without mocking `enrollmentDto.getStudentProfile()` into some mock `StudentProfile` object, the method returns **null**. This thing becomes trivial because this does not provide us with any confidence that the `studentProfileService.updateStudentProfile()` is really accepting something from the output of `enrollmentDto`.

It is recommended that we explicitly specify the mocking behavior of `enrollmentDto.getStudentProfile()`.

```java

    @Test
    public void processEnrollment_shouldUpdateStudentProfile() throws Exception {
        StudentProfile studentProfile = Mock(StudentProfile.class);
        EnrollmentDto enrollmentDto = mock(EnrollmentDto.class);
        when(enrollmentDto.getStudentProfile()).thenReturn(studentProfile);

        enrollmentService.processEnrollment(enrollmentDto);
        verify(studentProfileService, times(1)).updateStudentProfile(ArgumentMatchers.eq(studentProfile));
    }

```